include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -6052,6 +6053,19 @@ struct OrtApi {
   ORT_API2_STATUS(Node_GetAttributeByName, _In_ const OrtNode* node, _In_ const char* attribute_name,
                   _Outptr_ const OrtOpAttr** attribute);
 
+  /** \brief Get the OrtNode's 'TENSOR' attribute as an OrtValue.
+   *
+   * \param[in] node The OrtNode instance.
+   * \param[in] attribute The OrtOpAttr instance.
+   * \param[out] name Output parameter set to the attribute's name. The name is a null-terminated string.
+   *
+   * \snippet{doc} snippets.dox OrtStatus Return Value
+   *
+   * \since Version 1.23.
+   */
+  ORT_API2_STATUS(Node_GetTensorAttributeAsOrtValue, _In_ const OrtNode* node, _In_ const OrtOpAttr* attribute,
```

Comment:
Thought about changing the API to `OpAttr_GetTensorAttributeAsOrtValue` and make the OpAttr own the OrtValue instance. Then i need to make sure OpAttr is interchangeable with AttributeProto.